cmd/snap-update-ns/utils.go

```diff
@@ -654,18 +654,18 @@ func createWritableMimic(dir, neededBy string, as *Assumptions) ([]*Change, erro
 	return changes, nil
 }
 
-func applyProfile(snapName string, currentBefore, desired *osutil.MountProfile, as *Assumptions) (*osutil.MountProfile, error) {
+func applyProfile(up MountProfileUpdate, snapName string, currentBefore, desired *osutil.MountProfile, as *Assumptions) (*osutil.MountProfile, error) {
 	// Compute the needed changes and perform each change if
 	// needed, collecting those that we managed to perform or that
 	// were performed already.
-	changesNeeded := NeededChanges(currentBefore, desired)
+	changesNeeded := up.NeededChanges(currentBefore, desired)
 	debugShowChanges(changesNeeded, "mount changes needed")
 
 	logger.Debugf("performing mount changes:")
 	var changesMade []*Change
 	for _, change := range changesNeeded {
 		logger.Debugf("\t * %s", change)
-		synthesised, err := changePerform(change, as)
+		synthesised, err := up.PerformChange(change, as)
```

Comment:
it's a bit unclear why this was not Change.Perform(as).

There are other untouched changePerform in this file for which the same question is valid.

Reply:
This is not Change.Perform(as) because that part is not mockable and has complex filesystem interactions that also require root to operate.

All of the changePerform that are remaining in this branch exist because of `main_test.go` that I don't believe should be kept around, especially considering the full scope of the testing in the complete split branch.

In the original PR I didn't need to call MockChangePerform anyomore, at least not for testing the main algorithm.

Reply:
we are probably talking about different things, it's literally

https://github.com/snapcore/snapd/blob/master/cmd/snap-update-ns/change.go#L278

which can be mocked via MockChangePerform.   I don't know why it exists like that if it's meant only for tests.

Reply:
Ah, indeed. I must have missed that aspect originally.

Reply:
Fixed now.